test/e2e/conftest.py

```
@@ -124,3 +131,138 @@ def pytest_addoption(parser):
+    """Periodically capture Envoy memory stats from gateway pods.
+
+    Writes snapshots to $ARTIFACT_DIR/envoy_memory_snapshots.json.
+    Only active when --network-layer=gateway-api.
+    """
+    network = request.config.getoption("--network-layer", default="istio")
+    if network != "gateway-api":
+        yield
+        return
+
+    cli = os.environ.get("KUBE_CLI", "kubectl")
+    artifact_dir = os.environ.get("ARTIFACT_DIR", "/tmp")
```

Ruff (0.15.18)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)